### ISSUE TITLE
Display v.windowPeriod in dashboard header

### ISSUE THREAD

**Issue by philjb** [influxdata/ui#2529]
Thank you for suggesting an idea to improve InfluxData UI.

- Please ask usage questions on the Influx Community site.
  - https://community.influxdata.com/
- Please add a :+1: or comment on a similar existing feature request instead of opening a new one.
  - https://github.com/influxdata/ui/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+is%3Aclosed+sort%3Aupdated-desc+label%3A%22kind%2Ffeature+request%22+

**Proposal:**
The automatic windowPeriod is nice for automatic resolution binning but the value of it often matters for interpreting the results in a chart. 

This issue proposes showing the windowPeriod at the top of every dashboard e.g. along side the timerange perhaps.

**Current behavior:**
windowPeriod is opaque and not knowable

**Desired behavior:**
Always display the windowperiod used in a dashboard, update the display when it changes and otherwise make it exposed to customers.

**Alternatives considered:**
Create a specific cell to show the value. This is cumbersome and not always visible on screen (dashboard has many pages to scroll through).

```
import "generate"

generate.from(start: v.timeRangeStart, stop: v.timeRangeStart, count: 2, fn: (n) => 1)
  |> range(start: v.timeRangeStart, stop: v.timeRangeStop) // required to define windowPeriod var
  |> map(fn: (r) => ({r with _value: int(v: v.windowPeriod)/1000/1000/1000 }))
```

**Use case:**
Understanding data - allows determining units of charts.

**Comment by philjb** [comment]
The window period should also be on-screen visible in edit mode and in the explore page.

**Comment by stuartcarnie** [comment]
An additional suggestion is may we consider making the **Window Period** a combo box like the time range. It would default to "auto" to mirror the current behaviour, but allows the user to specify alternate values. Not a requirement but it might be beneficial if this list is customisable by the user?. I have been doing this myself by using a variable named `fixedWindowPeriod` with all my recent dashboards and making window period explicit:

![image](https://user-images.githubusercontent.com/52852/140192413-88a925d9-8601-4a02-95b5-76a3d6d538ce.png)

**Comment by kristinarobinson** [comment]
Talk to design to discuss how/where to display the window period.  @brandenTenbrink - can you take a look?

**Comment by Luckette** [comment]
I heard back from @mavarius, @taramk will take on the design effort for this. 

FYI @kristinarobinson

**Comment by AyeAndy** [comment]
@stuartcarnie @kristinarobinson @philjb by referring to dashboard and refresh, are you proposing to have some sort of windowperiod signifier on the right here?

![Screen Shot 2022-03-09 at 3 24 17 PM](https://user-images.githubusercontent.com/8049399/157556089-f5f2f58d-51f4-473e-90fa-e496fc563753.png)
?

**Comment by philjb** [comment]
Doesn't have to be on the right, but somewhere in the dashboard header is reasonable. The idea is to have it visible onscreen at all times for dashboards, cell edit mode, and explorer page edit.

**Comment by AyeAndy** [comment]
Gotcha. So this proposal assumes that the auto windowPeriod is same for all cells within the dashboard? Will there be any cases where each cells have different window periods?

**Comment by stuartcarnie** [comment]
@AyeAndy as @philjb said, somewhere on the dashboard that is visible is fine. I have replicated this by adding the `fixedWindowPeriod` variable to my queries, which causes it to show up on my dashboards.

<img width="702" alt="CleanShot 2022-03-10 at 10 37 42@2x" src="https://user-images.githubusercontent.com/52852/157557643-6d03dbf3-b683-4c99-ba93-2ff8fe555969.png">

I think it would be valuable to allow users to customise the list of values, if that is possible.

**Comment by philjb** [comment]
I think that is true (same for all cells) but I don't know. I believe it is calculated from the drop down time picker range. @hoorayimhelping or @kristinarobinson know or can find out who does I think.

**Comment by stuartcarnie** [comment]
The `v.windowPeriod` should be the same for all cells, otherwise the charts become difficult to interpret if we don't know the  intervals.

**Comment by AyeAndy** [comment]
@philjb @stuartcarnie Thanks for the quick response! Out of curiosity, do we have user feedback supporting prioritization on this feature? Specifically, would our user base like windowperiod on each cells or dashboards?

**Comment by stuartcarnie** [comment]
@AyeAndy No, though myself and  @philjb are the users requesting the feature. I don't see a valid reason it would be useful at the cell level, just as the date range would not be useful at the cell level.

**Comment by philjb** [comment]
I'm a fairly heavy user of our ui and I certainly would benefit from this feature. I don't know how the product team collects market feedback.

**Comment by philjb** [comment]
I was just asked 
> do you know what the window period is?

when troubleshooting a flux query.

**Comment by ALuckette** [comment]
This is somewhat implemented. For cells using Adaptive Zoom, the window period is displayed on the cell.
